### PR TITLE
Improve documentation for (kind .) stanza

### PR DESCRIPTION
**Pull request by pkel**
Issue: #1327

### CODE REVIEW

File: doc/dune-files.rst [hunk 1]
```diff
@@ -136,10 +136,9 @@ to use the :ref:`include_subdirs` stanza.
   advanced use only, by default you shouldn't set this option
 
 - ``(kind <kind>)`` is the kind of the library. The default is ``normal``, other
-  available choices are ``ppx_rewriter`` and ``ppx_deriver`` and must be set
-  when the library is intended to be used as a ppx rewriter or a ``[@@deriving
-  ...]`` plugin. The reason why ``ppx_rewriter`` and ``ppx_deriver`` are split
-  is historical and hopefully we won't need two options soon
+  available choices are ``ppx_deriver`` and ``ppx_rewriter``. ``ppx_deriver``
+  is used for ppx preprocessors using the ``ppx_deriving`` package and
```
Comment (emillon) [inline doc/dune-files.rst]: It's useful to explicitly name `ppx_deriving`, but I think that this should include something like "(or other packages handling [@@deriving])".

Comment (pkel) [reply doc/dune-files.rst]: I am still confused. I build a rewriter handling `[@@deriving something]` using `ppxlib`. Setting `(kind ppx_deriver)` causes problems. Using `(kind ppx_rewriter)` works fine. See [initial post](https://github.com/ocaml/dune/issues/1327#issue-363644006).

Comment (emillon) [reply doc/dune-files.rst]: Ah, sorry I missed that part. I am not sure what is the exact distinction then. @diml should be able to explain this better.

Comment (diml) [reply doc/dune-files.rst]: I just added a comment in the other ticket

**Comment by diml** [comment]
I looked a bit more into this. There are things we could do in dune to improve the situation but they would involve quite a bit of churn. The simplest thing to do is modify utop so that it implements its own ad-hoc logic to load ppx rewriters.

**Comment by pkel** [comment]
I close this PR, because my change of documentation does not improve the situation.

**Pull request closed without merging**